### DISCUSSION TITLE
Is it possible to insert content between the title and the body in a theme override?

### DISCUSSION THREAD

**Discussion by caseyamcl** [mkdocs/mkdocs#3170] [Q&A] [answered]
Hi, I'm a relatively new mkdocs user, and I'm wondering if it's possible to insert content between the title and body in the content template field. See below image:

![Screenshot from 2023-03-30 11-30-19](https://user-images.githubusercontent.com/53035/228888015-e3b33aef-86db-4fa2-9646-f5c9bd8690a3.png)

I see that there is a `{{ page.content }}` Jinja2 field, but it looks like it renders both the title and the body.

I'd like to create a custom theme override that has content in that location if possible.

**Comment by Andre601** [comment, marked as answer]
`{{ page.content }}` lists everything the markdown file contains.
Depending on the theme is stuff like the h1 header treated differently (mkdocs-material does that).

Also, depending on the theme could stuff like specific metadata be used in it.
Tho, if you know a bit of jinja could you override the theme and insert whatever you want.
For example, `{{ page.description }}` could be added  which would display whatever is provided in the `description` yaml frontmatter.

**Reply by caseyamcl** [reply]
Hi @Andre601 ! I guess you're right. mkdocs-material does separate the title from the body in the theme template. I find it confusing that they present it as a "hack" in the source code:

<https://github.com/squidfunk/mkdocs-material/blob/master/src/partials/content.html#L31>

Also, other mkdocs themes do not have a separate section for the title (for example, [Windmill](https://github.com/gristlabs/mkdocs-windmill/blob/master/mkdocs_windmill/base.html#L115) or [Cinder](https://github.com/chrissimpkins/cinder/blob/master/cinder/content.html)), so I made an assumption that it was part of the `{{ page.content }}`.

**Reply by kamilkrzyskow** [reply]
Looking at the commit history the hack was there since the beginning acting as a "nifty" design choice:
https://github.com/squidfunk/mkdocs-material/blob/420b841ecf6d79ec4ebc725b9a2ea82bb4217085/material/base.html#L82-L84

Some time ago there was an attempt to modify it, and make it act differently to maybe make it less "hacky" (?):
https://github.com/squidfunk/mkdocs-material/commit/b95dffa5cd7c32a443ab2c46d63252bc828409ec

But after years of public use, some people decided to design their docs around this hack. (either intentionally or by mistake) 😅, so you're right that the hack isn't a hack anymore ✌️ 

---

As for overriding the theme, I would like to point to my guide here to do it dynamically with hooks. Though, I'm currently working on some fixes to make the code less redundant and will update the guide soon™️.

https://github.com/squidfunk/mkdocs-material/discussions/5060

**Reply by caseyamcl** [reply]
Hi @kamilkrzyskow, thanks for the clarification. I understand why this is the case 100% better now.

Also, I appreciate the guide. It is exactly what I need to further customize our site the way that we need.
